pkg/ui/src/app.tsx

```diff
@@ -78,140 +78,114 @@ export const App: React.FC<AppProps> = (props: AppProps) => {
-                   component={normalizeConnectedComponent(StatementDetails)}/>
-          </Route>
-
-          { /* debug pages */}
-          <Route path="debug">
-            <IndexRoute component={Debug}/>
-            <Route path="redux" component={normalizeConnectedComponent(ReduxDebug)}/>
-            <Route path="chart" component={normalizeConnectedComponent(CustomChart)}/>
-            <Route path="enqueue_range" component={normalizeConnectedComponent(EnqueueRange)}/>
-          </Route>
-          <Route path="raft" component={Raft}>
-            <IndexRedirect to="ranges"/>
-            <Route path="ranges" component={normalizeConnectedComponent(RaftRanges)}/>
-            <Route path="messages/all" component={normalizeConnectedComponent(RaftMessages)}/>
-            <Route path={`messages/node/:${nodeIDAttr}`} component={normalizeConnectedComponent(RaftMessages)}/>
-          </Route>
-          <Route path="reports">
-            <Route path="problemranges" component={normalizeConnectedComponent(ProblemRanges)}>
-              <Route path={`:${nodeIDAttr}`} component={normalizeConnectedComponent(ProblemRanges)}/>
-            </Route>
-            <Route path="localities" component={normalizeConnectedComponent(Localities)}/>
-            <Route path="nodes">
-              <IndexRoute component={ normalizeConnectedComponent(Nodes) } />
-              <Route path="history" component={ normalizeConnectedComponent(ConnectedDecommissionedNodeHistory) } />
-            </Route>
-            <Route path="network" component={ normalizeConnectedComponent(Network) }>
-              <Route path={`:${nodeIDAttr}`} component={ normalizeConnectedComponent(Network) } />
-            </Route>
-            <Route path="settings" component={normalizeConnectedComponent(Settings)}/>
-            <Route path={`certificates/:${nodeIDAttr}`} component={normalizeConnectedComponent(Certificates)}/>
-            <Route path={`range/:${rangeIDAttr}`} component={normalizeConnectedComponent(Range)}/>
-            <Route path={`stores/:${nodeIDAttr}`} component={normalizeConnectedComponent(Stores)}/>
-          </Route>
-
-          { /* old route redirects */}
-          <Route path="cluster">
-            <IndexRedirect to="/metrics/overview/cluster"/>
-            <Redirect
-              from={`all/:${dashboardNameAttr}`}
-              to={`/metrics/:${dashboardNameAttr}/cluster`}
-            />
-            <Redirect
-              from={`node/:${nodeIDAttr}/:${dashboardNameAttr}`}
-              to={`/metrics/:${dashboardNameAttr}/node/:${nodeIDAttr}`}
-            />
-            <Route path="nodes">
-              <IndexRedirect to="/overview/list"/>
-              <Route path={`:${nodeIDAttr}`}>
-                <IndexRedirect to={`/node/:${nodeIDAttr}`}/>
-                <Redirect from="logs" to={`/node/:${nodeIDAttr}/logs`}/>
-              </Route>
+      <ConnectedRouter history={history}>
```

Comment:
Some of the tests in `app.spec.tsx` fail for me, can you take a look?

I noticed that the links in the Statements page in particular were broken too. The individual statements wouldn't link to their details pages and the dropdown wouldn't do anything.

Reply:
Fixed. The problem was in wrong ordering of routes, more general route path has been placed before more specific one.